### PR TITLE
removing builder-build from `build` and adding instructions how to build the builder

### LINKED ISSUE #4: Provide Developer Docs
Docs are missing for developers who want to contribute to builder itself.
How to build a new builder container on your local system? How to use it to build your own builder based distro?

### PR DESCRIPTION
**Pull request by MalteJ**
Fixes #4

### CODE REVIEW

File: build [hunk 1]
```diff
@@ -58,11 +58,6 @@ container_mount_opts=(
         -v "$PWD/$target_dir:/builder/.build"
 )
 
-if [ "$container_image" = localhost/builder ]; then
```
Comment (nkraetzschmar) [inline build:61]: This part of the build script is intended to speed up local dev / testing, by allowing to automatically rebuild the builder if any local changes were made.
It works if instead of downloading a release build script and placing it in the config directory, you instead sym link to the script inside a local copy of the builder repo.
We should not remove this

File: Dockerfile
Comment (nkraetzschmar) [inline Dockerfile:1]: why is this rename necessary?

File: build (hunk 1 above)
Comment (nkraetzschmar) [reply build:61]: we should probably document how this local dev / testing setup works though 😅

**Review by nkraetzschmar — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (MalteJ) [reply Dockerfile:1]: Docker wants to have a Dockerfile. Podman can work with both.

File: build (hunk 1 above)
Comment (MalteJ) [reply build:61]: but how do I use a locally built builder in a separate repo, that holds my features?
When I set the container image in the other repo to `localhost/builder` it tries to build the builder. Instead, it should use the builder that I have built before using `docker build -t localhost/builder .`

Comment (MalteJ) [reply build:61]: I don't want to build the builder in the same repo that hold my features. I want to have it separated.

**Review by Vincinator — APPROVED** [review]
It makes sense to me

File: build (hunk 1 above)
Comment (nkraetzschmar) [reply build:61]: The idea is to use a symlink to the build script inside the builder repo instead of a copy of the build script when doing local dev on the builder. That way the builder and the features are still separated, but the builder automatically detects that it is running in local dev mode and rebuilds itself iff necessary.
